### PR TITLE
More fixes for scanning Vue.js

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/216)

### CODE REVIEW

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 1]
```diff
@@ -35,6 +35,8 @@ import java.io.File
 import java.io.IOException
 
 abstract class GitBase : VersionControlSystem() {
+    override val movingRevisionNames = listOf("HEAD", "master")
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/Git.kt]: Do you plan the replace this with a list of the actual branches later?

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 2]
```diff
@@ -111,7 +113,7 @@ object Git : GitBase() {
 
             val workingTree = getWorkingTree(targetDir)
 
-            val revision = if (vcs.revision.isNotBlank()) {
+            val revision = if (vcs.revision.isNotBlank() && vcs.revision !in movingRevisionNames) {
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/Git.kt]: Repeating this condition in all VCS seems error prone, why not make a function in the parent class, something like `revisionIsValid`?

File: downloader/src/funTest/kotlin/BabelTest.kt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                    path = ""
+            )
+            val vcsFromUrl = VersionControlSystem.splitUrl(normalizeVcsUrl(vcsFromPackage.url))
+            val vcsMerged = vcsFromUrl.merge(vcsFromPackage)
+
+            val pkg = Package(
+                    packageManager = "NPM",
+                    namespace = "",
+                    name = "babel-cli",
+                    version = "6.26.0",
```
Comment (mnonnenmacher) [inline downloader/src/funTest/kotlin/BabelTest.kt]: You have to rebase to use an Identifier here.

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Git.kt]: No. My original idea was to really leave this hard-coded.

File: downloader/src/funTest/kotlin/BabelTest.kt (hunk 3 above)
Comment (sschuberth) [reply downloader/src/funTest/kotlin/BabelTest.kt]: Done.

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 2 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Git.kt]: Done.

**Pull request merged**